### ISSUE TITLE
portable vs non-portable headers

### ISSUE THREAD

**Issue by BradleySmall** [include-what-you-use/include-what-you-use#492]
In a large system cleaning up source code I am often times told to add

<sys/signal.h>
or <sys/limits.h>
and remove
<signal.h>
<limits.h>

It is not only these, but those come up often. 

This would conflict with warnings suggesting that is not the more portable code. And we are concerned with portability. 

Is there a way around this? Or is this by design?

**Comment by kimgr** [comment]
Yes, this is annoying, but it's an unfortunate consequence of the lack of metadata -- IWYU sees where the symbol is actually declared, and suggests that header. It can't see the difference between an indirectly included dependency that you should be including directly, and this. The mechanism for providing said metadata is mappings: you can configure IWYU to always use signal.h when sys/signal.h is suggested, for example. See docs on mappings for details.

**Comment by BradleySmall** [comment]
Thanks, I will look at that documentation for syntax and such.

**Comment by monouser7dig** [comment]
I run into the same problem cause I have Docker CI running IWYU but it won’t work on mac with the specified headers (or vice versa running IWYU on mac, it won’t compile in the Docker.

Maybe at least some heuristic can be added to mitigate the effects?

Thanks
